File: springboot-02-ioccontainer/src/main/java/com/linkedbear/springboot/assemble/a_module/TavernApplication.java
package com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional;

import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.config.TavernConfiguration;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

import java.util.stream.Stream;

public class TavernApplication {
    
    public static void main(String[] args) {
        AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(TavernConfiguration.class);
        Stream.of(ctx.getBeanDefinitionNames()).forEach(System.out::println);
        System.out.println("--------------------------");
//        Boss boss = ctx.getBean(Boss.class);
//        System.out.println(boss);
//        Map<String, Bartender> bartenders = ctx.getBeansOfType(Bartender.class);
//        bartenders.forEach((name, bartender) -> System.out.println(bartender));
    }
}

File: springboot-02-ioccontainer/src/main/kotlin/com/linkedbear/springboot/assemble/a_module/TavernApplication.kt
package com.linkedbear.springboot.assemble.a_module

import com.linkedbear.springboot.assemble.a_module.component.Bartender
import com.linkedbear.springboot.assemble.a_module.config.TavernConfiguration
import org.springframework.context.annotation.AnnotationConfigApplicationContext

class TavernApplication {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val ctx = AnnotationConfigApplicationContext(TavernConfiguration::class.java)
            ctx.beanDefinitionNames.forEach { println(it) }
            println("--------------------------------")
            val bartenders = ctx.getBeansOfType(Bartender::class.java)
            bartenders.forEach { (_, v) ->  println(v) }
        }
    }
}